azurelinuxagent/common/osutil/default.py

```diff
@@ -31,15 +31,18 @@
 import struct
 import sys
 import time
-from pwd import getpwall
+from pwd import getpwall # pylint: disable=ungrouped-imports
 
 import array
 
+#pylint: disable=R0801
```

Comment:
Why is this error disabled only for a few set of imports? Also I couldn't find a link for R0801 in the docs you added. Could you please add a link for it too, thanks!

Reply:
Also NIT: no space between # and pylint

Reply:
The symbol for this error code is `duplicate-code`. The reason why we have this within the codebase but not within the comments at the beginning of `*.pylintrc` is because I ended up suppressing `duplicate-code` globally. That we want to eventually fix this is captured within a `TODO` beside `duplicate-code` within the disabled section, and the links at the beginning of both files do in fact have an entry for `duplicate-code (R0801)`.

I agree, though, that having a line-item with the others at the start of the `*.pylintrc` files would be ideal.

Let's figure out the next PR we want approved, so I can make the change just one time, there.

Reply:
so if this code is suppressed globally we dont need this disable/enable here then do we?

Reply:
Not for suppression, no.

However, I would assume we want want to leave this comment in so we know where to fix it.

Reply:
But to fix it we would need to remove the global filter right? And once that is removed this line will start throwing errors in the linter right? Wouldn't that make this redundant?

Reply:
Yeah, that's true. I think just searching in the source code is preferred to running the linter and picking out the filename and line numbers one by one, though.

I feel like that's a good enough value add to keep these.

Reply:
I see. Is this the only place that particular error is raised? If not then you should add the same suppress line in every line that error would be raised by the linter if it was not globally suppressed. Whatever route you take at the end of the day, it should be consistent :)

Reply:
Right, everything is in fact consistent!

Reply:
So if this is the only place where its used, then why even add it to the global filter?

Reply:
It is not the only place it is used, sorry.

There are other places, and they are also marked. There seems to be a bug with `duplicate-code` that makes inline error suppression not work, hence the global disabling as well.

Reply:
Ahh alright, that makes sense! Thanks for adding it! I think just a description for this in the linters.rc file would be all I need. :)